packages/overlay/src/overlay-stack.ts

```diff
@@ -585,19 +593,27 @@ export class OverlayStack {
         if (this.preventMouseRootClose || event.defaultPrevented) {
             return;
         }
-        this.closeTopOverlay();
-        const overlaysToClose = [];
-        let root: HTMLElement | undefined = this.topOverlay?.root;
-        let overlay = parentOverlayOf(root);
-        while (root && overlay) {
-            overlaysToClose.push(overlay);
-            overlay = parentOverlayOf(root);
-            root = overlay?.root;
+        if (this.topOverlay?.interaction === 'hover') {
+            // Don't close the top overlay on click if it comes from a hover, it will be closed when the mouse is moved out
+            // But do close everything that came from a click since we just clicked elsewhere!
+            this.closeAllClickOverlays();
         }
-        if (overlay) {
-            overlaysToClose.push(overlay);
+        else {
+            this.closeTopOverlay();
+
+            const overlaysToClose = [];
+            let root: HTMLElement | undefined = this.topOverlay?.root;
+            let overlay = parentOverlayOf(root);
+            while (root && overlay) {
+                overlaysToClose.push(overlay);
+                overlay = parentOverlayOf(root);
+                root = overlay?.root;
+            }
+            if (overlay) {
+                overlaysToClose.push(overlay);
+            }
+            overlaysToClose.forEach((overlay) => this.hideAndCloseOverlay(overlay));
```

Comment:
I wonder about this code -- it's saying "close the top overlay, then if it's inside of other overlays, find them and close them too." I _think_ we could forgo this completely if we just close all click overlays, but I'm not sure if there will be funny edge cases...

Reply:
There are other types of overlays to take into account: https://github.com/adobe/spectrum-web-components/blob/main/packages/overlay/src/overlay-types.ts#L17-L24